pkg/kp/constants.go

```diff
@@ -1,38 +1,84 @@
 package kp
 
 import (
-	"strings"
+	"path"
+
+	"github.com/square/p2/pkg/types"
+	"github.com/square/p2/pkg/util"
 )
 
+// PodPrefix represents the top level of a subtree whose leaves are expected to
+// be pods, e.g. "hooks", "intent", "reality"
+type PodPrefix string
+
 const (
-	INTENT_TREE  string = "intent"
-	REALITY_TREE string = "reality"
-	HOOK_TREE    string = "hooks"
-	LOCK_TREE    string = "lock"
-	RC_TREE      string = "replication_controllers"
-	ROLL_TREE    string = "rolls"
+	INTENT_TREE  PodPrefix = "intent"
+	REALITY_TREE PodPrefix = "reality"
+	HOOK_TREE    PodPrefix = "hooks"
+	LOCK_TREE    string    = "lock"
 )
 
-func IntentPath(args ...string) string {
-	return strings.Join(append([]string{INTENT_TREE}, args...), "/")
-}
+var (
+	VALID_POD_PREFIXES = []PodPrefix{
+		INTENT_TREE,
+		REALITY_TREE,
+		HOOK_TREE,
+	}
+)
 
-func RealityPath(args ...string) string {
-	return strings.Join(append([]string{REALITY_TREE}, args...), "/")
-}
+func nodePath(podPrefix PodPrefix, nodeName string) (string, error) {
+	validPrefix := false
+	for _, prefix := range VALID_POD_PREFIXES {
```

Comment:
It depends what we want out of your modification. If we want to make it reasonably harder to operate on the wrong tree, then it's probably fine to create a few constants with well-defined values and then don't create any more. As a result, these constants can be implemented in a straightforward way. If you want to make it _impossible_ to use anything outside a couple constants, you can, but it complicates your design.

Personally, I'd shoot for the "reasonably hard to screw it up," and my comment was made to that end.

And If you really want to make it impossible to pick the wrong tree, you shouldn't make `VALID_POD_PREFIXES` public. I would also expect there to be a way to get the type system to prevent the construction of non-zero values of a public type.